api/src/main/java/io/opentelemetry/common/ImmutableKeyValuePairs.java

```diff
@@ -105,8 +106,8 @@ private static void quickSort(Object[] data, int leftIndex, int rightIndex) {
     for (int i = 0; i < data.length; i += 2) {
       Object key = data[i];
       Object value = data[i + 1];
-      if (key == null) {
-        logger.warning("Ignoring null key.");
+      if (key == null || "".equals(key)) {
+        logger.warning("Ignoring null/empty key.");
```

Comment:
logging here at warning seems like a recipe for a lot of log spam. I'm not sure if we should log this at all, but `warning` definitely seems too aggressive for something in the API.

Reply:
I also find it strange that we have a warning here. I don't think it adds anything, I will remove it